asyncpg/connection.py

```diff
@@ -1222,7 +1222,10 @@ def _on_release(self, stacklevel=1):
     def _drop_local_statement_cache(self):
         self._stmt_cache.clear()
 
-    def _drop_global_statement_cache(self):
+    def _drop_global_statement_cache(self, drop_type_cache=False):
+        if drop_type_cache:
+            # It is enough to clear the type cache only once.
+            self._protocol.get_settings().clear_type_cache()
```

Comment:
Type cache is global across connection to _all_ databases. Please make the invalidation database-specific.

Reply:
Am I mistaken or `self._protocol` is database-specific connection?
```asyncpg/connect_utils.py:261```

Reply:
Also:
```asyncpg/protocol/protocol.pyx:102```
```asyncpg/protocol/protocol.pyx:131```